lib/internal/fs/promises.js

```diff
@@ -357,13 +358,34 @@ async function copyFile(src, dest, mode) {
 
 // Note that unlike fs.open() which uses numeric file descriptors,
 // fsPromises.open() uses the fs.FileHandle class.
-async function open(path, flags, mode) {
+async function open(path, flag, mode) {
+  let options;
+  let parents;
+  // Handle case where 2nd argument is options-bag
+  if (typeof flag === 'object' && flag) {
+    options = flag;
+    flag = options.flag ?? 'r';
+    mode = options.mode ?? 0o666;
+    parents = options.parents;
+  }
   path = getValidatedPath(path);
-  const flagsNumber = stringToFlags(flags);
+  flag = stringToFlags(flag);
   mode = parseFileMode(mode, 'mode', 0o666);
-  return new FileHandle(
-    await binding.openFileHandle(pathModule.toNamespacedPath(path),
-                                 flagsNumber, mode, kUsePromises));
+  try {
+    return new FileHandle(
+      await binding.openFileHandle(pathModule.toNamespacedPath(path),
+                                   flag, mode, kUsePromises));
+  } catch (err) {
+    // Lazily create parent-subdirectories if they do not exist
+    if (err?.code === 'ENOENT' && parents && (flag & O_CREAT)) {
```

Comment:
I don' t think we need optional chaining here, as internals should never throw null or undefined.
```suggestion
    if (err.code === 'ENOENT' && parents && (flag & O_CREAT)) {
```